windows-gaming-driver/src/driver/libinput.rs

```diff
@@ -0,0 +1,81 @@
+use std::io;
+use std::collections::VecDeque;
+use std::cell::RefCell;
+
+use tokio_core::reactor::{Handle, PollEvented};
+use futures::{Async, Poll, Future, Stream};
+use futures::unsync::mpsc::{UnboundedSender, UnboundedReceiver, unbounded};
+use input::{Libinput, LibinputInterface};
+use input::event::Event;
+use input::event::pointer::PointerEvent;
+use libc::{open, close, ioctl, c_char, c_int, c_ulong, c_void};
+use mio::Ready;
+
+use super::my_io::MyIo;
+
+const EVIOCGRAB: c_ulong = 1074021776;
+
+unsafe extern "C" fn do_open(x: *const c_char, y: c_int, z: *mut c_void) -> c_int {
+    let fd = open(x, y);
```

Comment:
I disagree

what we really want here is a raw file descriptor

we are already in C land, there's no point in trying to act like this is rust code